openmc/material.py

```diff
@@ -136,6 +157,17 @@ def __init__(self, material_id=None, name='', temperature=None):
         # If specified, a list of table names
         self._sab = []
 
+        # Set density if provided
+        if (density is not None and density_units is None) or (
+            density is None and density_units is not None
+        ):
+            raise ValueError(
+                "Both density and density_units must be provided together."
+            )
+        elif density is not None and density_units is not None:
+            self.set_density(density_units, density)
```

Comment:
How do you feel about having `density_units` default to "g/cm3"? In that case, the first branch of the `if` could just be removed.